kv/new_memdb_arena.go

```diff
@@ -104,25 +166,52 @@ type nodeAllocator struct {
 }
 
 func (a *nodeAllocator) init() {
-	panic("todo")
+	a.nullNode = memdbNode{
+		up:    nullAddr,
+		left:  nullAddr,
+		right: nullAddr,
+		vptr:  nullAddr,
+	}
 }
 
 func (a *nodeAllocator) getNode(addr memdbArenaAddr) *memdbNode {
 	if addr.isNull() {
 		return &a.nullNode
 	}
 
-	panic("todo")
+	return (*memdbNode)(unsafe.Pointer(&a.blocks[addr.idx].buf[addr.off]))
 }
 
 func (a *nodeAllocator) allocNode(key Key) (memdbArenaAddr, *memdbNode) {
-	panic("todo")
+	nodeSize := 8*4 + 2 + 1 + len(key)
+	addr, mem := a.alloc(nodeSize, true)
+	n := (*memdbNode)(unsafe.Pointer(&mem[0]))
+	n.vptr = nullAddr
+	n.klen = uint16(len(key))
+	copy(n.getKey(), key)
+	return addr, n
 }
 
-func (a *nodeAllocator) freeNode(addr memdbArenaAddr) {}
+var testMode = false
+
+func (a *nodeAllocator) freeNode(addr memdbArenaAddr) {
+	if testMode {
```

Comment:
Can we make it a field of the allocator to avoid race?

Reply:
It will be set in test package's `init`. No race, and be enabled for all tests in other packages.